doc/notes/ch4_namespace.txt

```diff
@@ -0,0 +1,56 @@
+This note explains how the namespaces should be used in CH4.
+
+= Terms in this note:
+  1. Component: a set of data structures and/or functions that related
+     to a particular functionality.
+
+= MPID_ namespace
+  This is the abstract device interface (ADI). Usually, these names are
+  linked to their corresponding implementations in the MPIDI_ namespace.
+
+= MPIDI_ namespace
+  This is the namespace for internal functionalities of CH4.
+  Functionalities that is specific for CH4 should live in here. For
+  example, WorkQ and rankmap as they are device-independent.
```

Comment:
This paragraph is a bit confusing since you say that things that are specific to CH4 should be here, but then you say that they are device-independent. DO you mean netmod/shmod independent?

Reply:
Yes. I do mean netmod/shmmod independent.